src/callbacks_step/amr.jl

```diff
@@ -282,14 +282,24 @@ function (amr_callback::AMRCallback)(u_ode::AbstractVector, mesh::TreeMesh,
     coarsened_original_cells = Int[]
   end
 
- # Return true if there were any cells coarsened or refined, otherwise false
- has_changed = !isempty(refined_original_cells) || !isempty(coarsened_original_cells)
- if has_changed # TODO: Taal decide, where shall we set this?
-  # don't set it to has_changed since there can be changes from earlier calls
-  mesh.unsaved_changes = true
- end
-
- return has_changed
+  # Store whether there were any cells coarsened or refined
+  has_changed = !isempty(refined_original_cells) || !isempty(coarsened_original_cells)
+  if has_changed # TODO: Taal decide, where shall we set this?
+    # don't set it to has_changed since there can be changes from earlier calls
+    mesh.unsaved_changes = true
+  end
+
+  # Dynamically balance computational load by first repartitioning the mesh and then redistributing the cells/elements
+  @timeit_debug timer() "dynamic load balancing" if has_changed && mpi_isparallel()
```

Comment:
Something that came to me only now: We should have an option (i.e., a new constructor variable for the AMRCallback) to disable dynamic load balancing, e.g., `dynamic_load_balancing=true` (default: do it)